### PR TITLE
chore: Update Nodejs to 0.18

### PR DESCRIPTION
**Pull request by mutahhir**
- chore: Upgrade docker image to use Node 18
- chore: Add changelog entry
- chore: Update Node types version to 18
- chore: Update node engine requirement
- chore: Update node version check

### CODE REVIEW

File: packages/@cdktf/cli-core/templates/typescript/package.json [hunk 1]
```diff
@@ -17,6 +17,6 @@
     "upgrade:next": "npm i cdktf@next cdktf-cli@next"
   },
   "engines": {
-    "node": ">=16.0"
+    "node": ">=18.0"
```
Comment (xiehan) [inline packages/@cdktf/cli-core/templates/typescript/package.json:20]: I think this should be the first LTS version for v18, which was  `18.12.0`.

File: packages/cdktf-cli/src/bin/cmds/helper/check-environment.ts [hunk 2]
```diff
@@ -46,7 +46,7 @@ async function checkGoVersion() {
 
 async function checkNodeVersion() {
   const out = await getNodeVersion();
-  throwIfLowerVersion("Node.js", "16.13.0", out);
+  throwIfLowerVersion("Node.js", "18.17.1", out);
```
Comment (xiehan) [inline packages/cdktf-cli/src/bin/cmds/helper/check-environment.ts:49]: I think this should be the first LTS version for v18, which was  `18.12.0`.

File: examples/typescript/google/package.json [hunk 3]
```diff
@@ -16,15 +16,15 @@
     "upgrade:next": "npm i cdktf@next cdktf-cli@next"
   },
   "engines": {
-    "node": ">=16.0"
+    "node": ">=18.0"
```
Comment (xiehan) [inline examples/typescript/google/package.json:19]: I think this should be the first LTS version for v18, which was  `18.12.0`.

File: examples/typescript/kubernetes/package.json [hunk 4]
```diff
@@ -16,15 +16,15 @@
     "upgrade:next": "npm i cdktf@next cdktf-cli@next"
   },
   "engines": {
-    "node": ">=16.0"
+    "node": ">=18.0"
```
Comment (xiehan) [inline examples/typescript/kubernetes/package.json:19]: I think this should be the first LTS version for v18, which was  `18.12.0`.

File: examples/typescript/ucloud/package.json [hunk 5]
```diff
@@ -16,15 +16,15 @@
     "upgrade:next": "npm i cdktf@next cdktf-cli@next"
   },
   "engines": {
-    "node": ">=16.0"
+    "node": ">=18.0"
```
Comment (xiehan) [inline examples/typescript/ucloud/package.json:19]: I think this should be the first LTS version for v18, which was  `18.12.0`.

File: examples/typescript/vault/package.json [hunk 6]
```diff
@@ -19,15 +19,15 @@
     "upgrade:next": "npm i cdktf@next cdktf-cli@next"
   },
   "engines": {
-    "node": ">=16.0"
+    "node": ">=18.0"
```
Comment (xiehan) [inline examples/typescript/vault/package.json:22]: I think this should be the first LTS version for v18, which was  `18.12.0`.

File: examples/typescript/backends/s3/package.json [hunk 7]
```diff
@@ -16,15 +16,15 @@
     "upgrade:next": "npm i cdktf@next cdktf-cli@next"
   },
   "engines": {
-    "node": ">=16.0"
+    "node": ">=18.0"
```
Comment (xiehan) [inline examples/typescript/backends/s3/package.json:19]: I think this should be the first LTS version for v18, which was  `18.12.0`.

File: examples/typescript/backends/remote/package.json [hunk 8]
```diff
@@ -16,15 +16,15 @@
     "upgrade:next": "npm i cdktf@next cdktf-cli@next"
   },
   "engines": {
-    "node": ">=16.0"
+    "node": ">=18.0"
```
Comment (xiehan) [inline examples/typescript/backends/remote/package.json:19]: I think this should be the first LTS version for v18, which was  `18.12.0`.

File: examples/typescript/backends/gcs/package.json [hunk 9]
```diff
@@ -16,15 +16,15 @@
     "upgrade:next": "npm i cdktf@next cdktf-cli@next"
   },
   "engines": {
-    "node": ">=16.0"
+    "node": ">=18.0"
```
Comment (xiehan) [inline examples/typescript/backends/gcs/package.json:19]: I think this should be the first LTS version for v18, which was  `18.12.0`.

File: examples/typescript/backends/cloud/package.json [hunk 10]
```diff
@@ -18,15 +18,15 @@
     "upgrade:next": "npm i cdktf@next cdktf-cli@next"
   },
   "engines": {
-    "node": ">=16.0"
+    "node": ">=18.0"
```
Comment (xiehan) [inline examples/typescript/backends/cloud/package.json:21]: I think this should be the first LTS version for v18, which was  `18.12.0`.

File: examples/typescript/backends/azurerm/package.json [hunk 11]
```diff
@@ -16,15 +16,15 @@
     "upgrade:next": "npm i cdktf@next cdktf-cli@next"
   },
   "engines": {
-    "node": ">=16.0"
+    "node": ">=18.0"
```
Comment (xiehan) [inline examples/typescript/backends/azurerm/package.json:19]: I think this should be the first LTS version for v18, which was  `18.12.0`.

File: examples/typescript/azure/package.json [hunk 12]
```diff
@@ -16,15 +16,15 @@
     "upgrade:next": "npm i cdktf@next cdktf-cli@next"
   },
   "engines": {
-    "node": ">=16.0"
+    "node": ">=18.0"
```
Comment (xiehan) [inline examples/typescript/azure/package.json:19]: I think this should be the first LTS version for v18, which was  `18.12.0`.

File: examples/typescript/azure-service-bus-queue-trigger/package.json [hunk 13]
```diff
@@ -15,7 +15,7 @@
     "upgrade:next": "npm i cdktf@next cdktf-cli@next"
   },
   "engines": {
-    "node": ">=16.0"
+    "node": ">=18.0"
```
Comment (xiehan) [inline examples/typescript/azure-service-bus-queue-trigger/package.json:18]: I think this should be the first LTS version for v18, which was  `18.12.0`.

**Review by DanielMSchmidt — APPROVED** [review]

**Pull request closed without merging**